### DISCUSSION TITLE
Consider supporting React 17

### DISCUSSION THREAD

**Discussion by SimenB** [TanStack/query#8458] [Ideas]
👋 

React compiler supports targeting React 17, so we have begun introducing it into our legacy code base. This code base is stuck on React 17 since it uses `reflux`. We have started to migrate to MobX instead, but since MobX is not compatible with the React compiler, we have instead decided to attempt to migrate to `react-query`. However, v5 only supports React 18+, so we have to use v4.

Would it be possible to extend support in v5 down to React 17, to match what React Compiler can target?

https://react.dev/learn/react-compiler#using-react-compiler-with-react-17-or-18

**Comment by TkDodo** [comment]
Sadly no because we use useSyncExternalStore which only exists in 18, and the shim for v17 was very troublesome for us in terms of bundling
